node/core/dispute-coordinator/src/import.rs

```diff
@@ -477,6 +477,35 @@ impl ImportResult {
 		self.is_freshly_concluded_against() || self.is_freshly_concluded_for()
 	}
 
+	// Returns the number of invalid own votes from the state provided. This function is a helper for
+	// `has_fresh_byzantine_threshold_against`
+	fn get_own_invalid_votes<T>(state: &CandidateVoteState<T>) -> usize {
+		state
+			.own_votes()
+			.map(|votes| {
+				votes.iter().fold(0, |acc, (_, (vote, _))| match vote {
+					DisputeStatement::Invalid(_) => acc + 1,
+					_ => acc,
+				})
+			})
+			.unwrap_or(0)
+	}
+
+	/// Whether or not the invalid vote count for the dispute went beyond the byzantine threshold
+	/// after the last import
+	pub fn has_fresh_byzantine_threshold_against(&self, byzantine_threshold: usize) -> bool {
+		// Total number of imported invalid votes + our own invalid votes
+		let imported_invalid_votes =
+			self.imported_invalid_votes() as usize + Self::get_own_invalid_votes(self.new_state());
+		// The number of invalid votes + our own invalid votes before the import
+		let old_invalid_votes = (self.imported_invalid_votes() as usize)
+			.saturating_sub(self.new_invalid_voters().len()) +
+			Self::get_own_invalid_votes(self.old_state());
+
+		old_invalid_votes as usize <= byzantine_threshold &&
+			imported_invalid_votes as usize > byzantine_threshold
```

Comment:
Why would we only check for newly imported invalid votes?

Reply:
I want `has_fresh_byzantine_threshold_against()` to return `true` only once - when `byzantine_threshold` is reached. So I check if the invalid votes before import were below (or equal) the `byzantine_threshold` and at the same time they are above it after the import.
But you probably mean something else?

Reply:
Yeah, I understand that you want to trigger the message once the threshold is passed and not on every import, but you are comparing newly imported votes to the byzantine threshold - so any previously known votes are not taken into account. This would only trigger if we imported f+1 invalid votes all at once.

Reply:
Got it, `self.imported_invalid_votes()` returns the votes from the last import, not all votes. This is totally messed up.

Reply:
This should be fixed now, but I have to add more tests. The current ones didn't catch that error.